apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -0,0 +1,299 @@
+    const resourcesLocalizations = await loadAllLocalizedVersionsOfMeterResources({ context })
+
+    // let's load meters page by page
+    const state = {
+        pageSize: 100,
+        offset: 0,
+        hasMore: true,
+        processedMeters: 0,
+        processedPages: 0,
+        metersWithoutReadings: 0,
+        metersWithoutReadingsAndWithResidents: 0,
+        metersWithExpiredVerificationDate: 0,
+        startTime: dayjs(),
+    }
+    log(`Start proceeding meters page by page ${ state.startTime }`, '-')
+
+    do {
+        // log each 10 pages
+        if (state.processedPages % 10 === 0) {
+            log(`Processing page ${ state.processedPages + 1 }`)
+        }
+
+        // read all required data
+        const meters = await readMetersPage({
+            context,
+            offset: state.offset,
+            pageSize: state.pageSize,
+        })
+        const meterReadings = await readMeterReadings({ context, meters })
+        const organizations = await readOrganizations({ context, meters })
+
+        // right join data to metersPage
+        const metersWithoutReadings = rightJoin(
+            meters,
+            meterReadings,
+            (meter, item) => meter.id === item.meter.id,
+            (meter, readings) => ({ meter, readings })
+        )
+            .filter(({ readings }) => readings.length === 0)
+            .map(({ meter }) => meter)
+
+        // right join organizations
+        const metersWithoutPeriod = rightJoin(
+            metersWithoutReadings,
+            organizations,
+            (meter, organization) => meter.organization.id === organization.id,
+            (meter, organization) => {
+                /**
+                 * Detect message language
+                 * Use DEFAULT_LOCALE if organization.country is unknown
+                 * (not defined within @condo/domains/common/constants/countries)
+                 */
+                const lang = get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+                const period = null // TODO calculate this prop after implementation submit period in organisation
+                return { ...meter, period, lang }
+            }
+        ).filter(({ period }) => period == null) // pick meters without organisation default period
+        state.metersWithoutReadings += metersWithoutPeriod.length
+
+        // Join residents
+        const metersWithResident = await joinResidentsToMeters({ context, meters: metersWithoutPeriod })
+        state.metersWithoutReadingsAndWithResidents += metersWithResident.length
+
+        // Send message with specific unique key for set up readings
+        await sendMessagesForSetUpReadings({
+            context, metersWithResident,
+        })
+
+        // Filter meters with expired next verification date
+        const expiredVerificationMetersWithResident = metersWithResident.filter(({ meter }) => {
+            const nextVerificationDate = dayjs(meter.nextVerificationDate)
```

Comment:
usually we have a `null` for `nextVerificationDate`

Reply:
You need to select meters with that date or assume the it may be null

Reply:
ok